packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1881,19 +1881,19 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     newRoute.install(_currentOverlayEntry);
     _history[index] = newRoute;
     newRoute.didPush().whenCompleteOrCancel(() {
-      // The old route's exit is not animated. We're assuming that the
-      // new route completely obscures the old one.
       if (mounted) {
         oldRoute
           ..didComplete(result ?? oldRoute.currentResult)
           ..dispose();
+        if (index > 0) {
+          _history[index - 1].didChangeNext(newRoute);
```

Comment:
When the push animation has completed, the route history may look differently and the index computed before the transition started may not be correct anymore.

Please add a test for this scenario as well to make sure that the correct routes are notified in this case.

Reply:
I have made a change to remove the use of the index value, instead reusing reference to the relevant route itself.
However, I'm struggling to write a test for this - if I push a route during the replace animation, the index used previously is still correct (test won't fail before my amendment). If I try pop a route, the popped route will be the one that this PR affects (nothing left to test). Any suggestions?

Reply:
Could you use removeRoute [1] to remove something lower in the stack? That should change indexes.

[1] https://master-api.flutter.dev/flutter/widgets/NavigatorState/removeRoute.html

Reply:
That did it - thank you. I have added the test as requested. I also had to refrain from using pop() to test the secondaryAnimation states as pop() re-creates the animation proxies.